datafusion/expr/src/signature.rs

```diff
@@ -42,10 +42,10 @@ pub enum TypeSignature {
     /// arbitrary number of arguments of an common type out of a list of valid types
     // A function such as `concat` is `Variadic(vec![DataType::Utf8, DataType::LargeUtf8])`
     Variadic(Vec<DataType>),
-    /// arbitrary number of arguments of an arbitrary but equal type
+    /// arbitrary number of arguments of an equal type
     // A function such as `array` is `VariadicEqual`
     // The first argument decides the type used for coercion
-    VariadicEqual,
+    VariadicEqual(Vec<DataType>),
```

Comment:
If the types are all equal, what is the purpose of storing a Vec of them? or maybe the Vec is the list of valid types?

It also looks like the code special cases when there are no types specified to mean "any type is allowed" -- can we please explicitly mention that in the docs as well? Or maybe add a new variant (`VariadicEqualSpecific`?)